doc/model.rst

```diff
@@ -0,0 +1,739 @@
+*****
+Model
+*****
+========
+Overview
+========
+SmartSim ``Model`` objects enable users to execute computational tasks in an
+``Experiment`` workflow, such as launching compiled applications,
+running scripts, or performing general computational operations. ``Models`` can be launched with
+other SmartSim entities and infrastructure to build AI-enabled workflows.
+``Model`` objects can leverage ML capabilities by utilizing the SmartSim client (:ref:`SmartRedis<dead_link>`)
```

Comment:
Guessing the `dead_link`s will be fixed

Reply:
Yes! After merge into the feature branch